File: backend/src/main/java/com/jobpulse/service/InterviewService.java
package com.jobpulse.service;

import com.jobpulse.dto.request.InterviewRequest;
import com.jobpulse.dto.response.InterviewResponse;

import java.util.List;

public interface InterviewService {
    InterviewResponse scheduleInterview(Long userId, Long applicationId, InterviewRequest request);
    List<InterviewResponse> getInterviewsByApplication(Long userId, Long applicationId);
    InterviewResponse updateInterview(Long userId, Long interviewId, InterviewRequest request);
    void deleteInterview(Long userId, Long interviewId);
}

File: frontend/src/services/interviewService.js
import api from './api';

export const interviewService = {
  scheduleInterview: async (applicationId, data) => {
    const response = await api.post(`/applications/${applicationId}/interviews`, data);
    return response.data;
  },
  getInterviews: async (applicationId) => {
    const response = await api.get(`/applications/${applicationId}/interviews`);
    return response.data;
  },
  updateInterview: async (interviewId, data) => {
    const response = await api.put(`/interviews/${interviewId}`, data);
    return response.data;
  },
  deleteInterview: async (interviewId) => {
    await api.delete(`/interviews/${interviewId}`);
  }
};
